Traceback (most recent call last):
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/tconnolly/opt/miniconda3/envs/ms263-24/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
plt.figure()
plt.xlim([-1,1])
plt.ylim([-1,1])

plt.xlabel('PC1 loading')
plt.ylabel('PC2 loading')

variables = ['OD','BD','LTD','W']
for i,txt in enumerate(variables):
    plt.plot([0,A[i,0]],[0,A[i,1]],'-')
    plt.text(A[i,0],A[i,1],txt)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[12], line 10[0m
[1;32m      8[0m variables [38;5;241m=[39m [[38;5;124m'[39m[38;5;124mOD[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124mBD[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124mLTD[39m[38;5;124m'[39m,[38;5;124m'[39m[38;5;124mW[39m[38;5;124m'[39m]
[1;32m      9[0m [38;5;28;01mfor[39;00m i,txt [38;5;129;01min[39;00m [38;5;28menumerate[39m(variables):
[0;32m---> 10[0m     plt[38;5;241m.[39mplot([[38;5;241m0[39m,[43mA[49m[i,[38;5;241m0[39m]],[[38;5;241m0[39m,A[i,[38;5;241m1[39m]],[38;5;124m'[39m[38;5;124m-[39m[38;5;124m'[39m)
[1;32m     11[0m     plt[38;5;241m.[39mtext(A[i,[38;5;241m0[39m],A[i,[38;5;241m1[39m],txt)

[0;31mNameError[0m: name 'A' is not defined

